src/main/java/org/opensearch/ad/ml/ModelManager.java

```diff
@@ -475,6 +475,8 @@ public void stopModel(String detectorId, String modelId, ActionListener<Void> li
     /**
      * Permanently deletes models hosted in memory and persisted in index.
      *
+     * TODO: confirm with Kaituo, when will we call this clear method, only see when call this method stop realtime job
```

Comment:
@kaituo , I only see this method called in `DeleteModelTransportAction` , which will be executed when stop detector. If detector still running, just AD job coordinating node changed, then these models will not be cleared. 

Should we clean up all models in cache for detector if its AD job coordinating node changed?

Reply:
Discussed with Kaituo offline, if node added/removed, we can check if model owning node can match local node or not. If can't match, we can delete model cache